build-info-extractor-npm/src/main/java/org/jfrog/build/extractor/npm/extractor/NpmBuildInfoExtractor.java

```diff
@@ -190,17 +236,26 @@ static boolean isJsonOutputRequired(List<String> commandArgs) {
         return commandArgs.contains("--json=true");
     }
 
-    /**
-     * npm install scope can be set by "--production" or "--only={prod[uction]|dev[elopment]}" flags.
-     *
-     * @param npmrcProperties - The results of 'npm config list' command.
-     */
-    private void setScope(Properties npmrcProperties) {
-        String only = npmrcProperties.getProperty("only");
-        if (StringUtils.startsWith(only, "prod") || Boolean.parseBoolean(npmrcProperties.getProperty("production"))) {
-            npmPackageInfo.setScope(NpmScope.PRODUCTION.toString());
-        } else if (StringUtils.startsWith(only, "dev")) {
-            npmPackageInfo.setScope(NpmScope.DEVELOPMENT.toString());
+    private void setTypeRestriction(String key, String value) {
```

Comment:
This function has complicated logic. I think it deserves a comprehensive unit test.